js/ui/status/keyboard.js

```diff
@@ -480,6 +484,51 @@ const InputSourceManager = new Lang.Class({
             this._updateMruSettings();
     },
 
+    get passwordModeEnabled() {
+        return this._passwordModeEnabled;
+    },
+
+    set passwordModeEnabled(enable) {
+        this._passwordModeEnabled = enable;
```

Comment:
I don't see why you would need to call `this._syncPasswordMode()` if `enable == this._passwordModeEnabled` already.

Reply:
We do need to reset the kbd layout to the previous one. 

passwordModeEnabled(true):
- store the current kbd layout
- lookup a possible latin layout and switch to it, or switch to 'en'

passwordModeEnabled(false):
- switch back to stored kbd layout
- reset flags etc

Reply:
I understand that, but I don't think you need to do any of that work when setting the same state again, do you?

Reply:
Eeh, yeah - sorry for misreading. I have it now.